Fix the tooltips on mobile devices

fritzmg opened the pull request:
Fixes #6382 

Same as #7583 for Contao 5.4+

fritzmg committed:
Fix tooltips on mobile devices

aschempp reviewed (commented):
Looks like we never unregister those event listeners? That could be a massive problem with Turbo...
Not sure if you wanna fix that in this PR as well?

fritzmg committed:
remove document event listeners again

fritzmg committed:
merge 5.4

fritzmg commented:
I have at least added unregistering of the global document touchstart listener in this PR now.

fritzmg commented:
> Looks like we never unregister those event listeners? That could be a massive problem with Turbo...
> Not sure if you wanna fix that in this PR as well?

I don't think we have to fix anything there. The eventlisteners are registered only once and only to the element itself. You don't have to remove eventlisteners on elements that will be removed from the DOM.

It's different for the global `touchstart` listener that I added - which is now fixed anyway.

leofeyer committed:
Merge branch '5.4' into fix-tips-on-mobile-54

leofeyer commented on core-bundle/assets/controllers/tooltip-controller.js:
```diff
@@ -12,6 +12,14 @@ export default class extends Controller {
             this.observer.disconnect();
         }
+
+        Object.entries(this.selectors).forEach(([selector, options]) => {
```
<img width="869" alt="" src="https://github.com/user-attachments/assets/f32bb450-faa0-4485-b64a-577e19761246">

fritzmg replied:
Fixed in 4aef7c2199

fritzmg committed:
only iterate over keys

leofeyer commented:
Thank you @fritzmg.
